integration-tests/deployment/ccip/test_assertions.go

```diff
@@ -14,9 +18,128 @@ import (
 	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 
 	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/offramp"
 )
 
+func ConfirmGasPriceUpdatedForAll(
+	t *testing.T,
+	e deployment.Environment,
+	state CCIPOnChainState,
+	startBlocks map[uint64]*uint64,
+) {
+	var wg errgroup.Group
+	for src, srcChain := range e.Chains {
+		for dest, dstChain := range e.Chains {
+			if src == dest {
+				continue
+			}
+			srcChain := srcChain
+			dstChain := dstChain
+			wg.Go(func() error {
+				var startBlock *uint64
+				if startBlocks != nil {
+					startBlock = startBlocks[srcChain.Selector]
+				}
+				return ConfirmGasPriceUpdated(
+					t,
+					dstChain,
+					state.Chains[srcChain.Selector].FeeQuoter,
+					*startBlock,
+				)
+			})
+		}
+	}
+	require.NoError(t, wg.Wait())
+}
+
+func ConfirmGasPriceUpdated(
+	t *testing.T,
+	dest deployment.Chain,
+	srcFeeQuoter *fee_quoter.FeeQuoter,
+	startBlock uint64,
+) error {
+	it, err := srcFeeQuoter.FilterUsdPerUnitGasUpdated(&bind.FilterOpts{
+		Context: context.Background(),
+		Start:   startBlock,
+	}, []uint64{dest.Selector})
+
+	require.NoError(t, err)
+	require.True(t, it.Next())
+	require.NotEqual(t, InitialGasPrice, it.Event.Value)
+	return nil
+}
+
+func ConfirmTokenPriceUpdatedForAll(
+	t *testing.T,
+	e deployment.Environment,
+	state CCIPOnChainState,
+	startBlocks map[uint64]*uint64,
+) {
+	var wg errgroup.Group
+	for _, chain := range e.Chains {
+		chain := chain
+		wg.Go(func() error {
+			var startBlock *uint64
+			if startBlocks != nil {
+				startBlock = startBlocks[chain.Selector]
+			}
+			linkAddress := state.Chains[chain.Selector].LinkToken.Address()
+			wethAddress := state.Chains[chain.Selector].Weth9.Address()
+			tokenToPrice := make(map[common.Address]*big.Int)
+			tokenToPrice[linkAddress] = InitialLinkPrice
+			tokenToPrice[wethAddress] = InitialWethPrice
+			return ConfirmTokenPriceUpdated(
+				t,
+				chain,
+				state.Chains[chain.Selector].FeeQuoter,
+				*startBlock,
+				tokenToPrice,
+			)
+		})
+	}
+	require.NoError(t, wg.Wait())
+}
+
+func ConfirmTokenPriceUpdated(
+	t *testing.T,
+	chain deployment.Chain,
+	feeQuoter *fee_quoter.FeeQuoter,
+	startBlock uint64,
+	tokenToInitialPrice map[common.Address]*big.Int,
+) error {
+	tokens := make([]common.Address, 0, len(tokenToInitialPrice))
+	for token := range tokenToInitialPrice {
+		tokens = append(tokens, token)
+	}
+	it, err := feeQuoter.FilterUsdPerTokenUpdated(&bind.FilterOpts{
+		Context: context.Background(),
+		Start:   startBlock,
+	}, tokens)
+	require.NoError(t, err)
+	for it.Next() {
+		token := it.Event.Token
+		initialValue, ok := tokenToInitialPrice[token]
```

Comment:
I guess you can inline it, `initialValue` is used only in this scope so it should be fine 
```go
if initialValue, ok := tokenToInitialPrice[token]; ok {
	    require.Contains(t, tokens, token)
            require.NotEqual(t, initialValue, it.Event.Value)
}